docs/api.md

```diff
@@ -1116,6 +1131,9 @@ puppeteer.launch().then(async browser => {
 
 ```
 
+#### page.extensionContexts()
```

Comment:
I don't like restricting us to extensions. Execution contexts can exist and might be handy in non-chromium embedders, e.g. in headless. 

From this point of view, the ideal API would be like this:
```js
// ExecutionContext lifecycle
page.on('executioncontextcreated');
page.on('executioncontextdestroyed');

// Querying  contexts
const contexts = frame.executionContexts();
const context = frame.defaultExecutionContext();

// Inspecting context
context.name(); // returns extension name, if any
context.isDefault(); // returns `true` if this is a default execution context
```

The only downside here is that we already have `frame.executionContext()` instead of `frame.defaultExecutionContext()`. I think we can keep it like this for now, and rename it later on in the API cleanup for v2.0.

Reply:
An api that contains all execution contexts is reasonable, but id still want helper methods specifically for extension contexts.

I’m wary of releasing an api that will today be effectively only extension contexts, but in the future might contain some more exotic contexts.

Reply:
> An api that contains all execution contexts is reasonable, but id still want helper methods specifically for extension contexts.

I'd move slowly into this direction. For the first-class extension support, we'll need to have an `Extension` class - there you might have methods like `extension.contentScript(frame)` and others.